tests/prague/eip7702_set_code_tx/test_set_code_txs.py

```diff
@@ -3460,3 +3460,86 @@ def test_authorization_reusing_nonce(
             sender: Account(nonce=1),
         },
     )
+
+
+@pytest.mark.parametrize(
+    "set_code_type",
+    list(AddressType),
+    ids=lambda address_type: address_type.name,
+)
+@pytest.mark.parametrize(
+    "self_sponsored",
+    [True, False],
+)
+@pytest.mark.execute(pytest.mark.skip(reason="Requires contract-eoa address collision"))
+def test_set_code_from_account_with_non_delegating_code(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    set_code_type: AddressType,
+    self_sponsored: bool,
+):
+    """
+    Test that a transaction is correctly rejected if the sender account has a non-delegating code
+    set.
+    """
+    sender = pre.fund_eoa()
+    if self_sponsored:
+        auth_signer = sender
```

Comment:
maybe no need to change the meaning of auth_signer here conditionally. it is difficult to read the code keeping in mind that auth_signer is a variable. better be constant